connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1061,6 +1068,81 @@ class SparkConnectPlanner(session: SparkSession) {
     }
   }
 
+  private def transformNumericAggregate(agg: proto.Aggregate): LogicalPlan = {
```

Comment:
Is that duplication really necessary? What exactly is the difference on the server side for the numeric aggregation? Do we have this special handling in Catalyst for this special case?

Reply:
I believe it's necessary to have a separate method for it, actually at first I attempted to implement it in existing `transformAggregate` then I found that the two paths are too different:
1, for the regular agg, we build the logical plan;
2, for the numeric agg, we have to use the dataframe API to avoid doing the analysis by myself.

Reply:
Can't you just let the analysis fail? It will fail anyways if you try to aggregate on non numeric columns.